src/classes/GE_GiftEntryController.cls

```diff
@@ -65,16 +65,17 @@ public with sharing class GE_GiftEntryController {
             GE_PaymentServices.ResponseWrapper response = paymentServices.createTransaction();
 
             return JSON.serialize(response);
-        } catch (Exception ex) {
-            Boolean isCalloutException = ex instanceof System.CalloutException;
-            Boolean hasTimedout = isCalloutException && ex.getMessage().contains(TIMED_OUT);
+        } catch (CalloutException ex) {
+            Boolean hasTimedout = ex.getMessage().contains(TIMED_OUT);
 
             if (hasTimedout) {
                 return getPurchaseCallTimeoutResponse(ex, dataImportRecordId);
             } else {
-                String JSONExceptionData = ERR_ExceptionData.createExceptionWrapperJSONString(ex);
-                throw returnDmlException(JSONExceptionData);
+                return null;
```

Comment:
@dfuller114 What's the logic behind returning "null" here? What happens from the user perspective if we hit this line? This seems like any CalloutException that's not a timeout gets caught and dead-ends unhelpfully here, rather than returning to the user in a way that provides a helpful message (that said, I didn't manage to actually hit any weird behavior here)

Reply:
@ptewson-sfdo looking back at this, I don't like returning `null` since it's an anti-pattern. I just pushed up a change for this where we just conditionally set a `timeoutResponse` string variable if that `TIMED_OUT` boolean is true; otherwise, the string is just an empty string that gets returned.